pkg/httpcodec/request.go

```diff
@@ -0,0 +1,62 @@
+package httpcodec
```

Comment:
Why is `httpcodec` not part of the SDK?

Reply:
Will think about it.

It actually depends on where we want to have the transport protocol specifics to live. The idea was to make the SDK agnostic of these details. But I also think the SDK should provide tooling to deal with the different transports :thinking: .

Reply:
These should be in the SDK.

Making the SDK agnostic of things is a byproduct. The SDK is not "an interface", it's a "Software Development Kit".

Ask yourself: "Will both nodie's gateway-server-kit and Grove's portal need it?" If so, it should be in the SDK.

Reply:
Worth adding ☝️ to the docs